addons/website_sale_stock/tests/test_website_sale_stock_product_warehouse.py

```diff
@@ -1,41 +1,31 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+from odoo.addons.website.tools import MockRequest
 from odoo.addons.website_sale.tests.test_website_sale_product_attribute_value_config import TestWebsiteSaleProductAttributeValueConfig
 
 
 class TestWebsiteSaleStockProductWarehouse(TestWebsiteSaleProductAttributeValueConfig):
 
-    def test_get_combination_info(self):
-        """ Checked that correct product quantity is shown in website according to the warehouse
-        which is set in current website.
-            - Create two warehouse
-            - Create two stockable products
-            - Update quantity of Product A in Warehouse 1
-            - Update quantity of Product B in Warehouse 2
-            - Set Warehouse 1, Warehouse 2 or none in website and:
-                - Check available quantity of Product A and Product B in website
-        When the user doesn't set any warehouse, the module should still select a default one
-        """
-        # Create two warehouses
-        warehouse_1 = self.env['stock.warehouse'].create({
-            'name': 'Warehouse 1',
-            'code': 'WH1'
-        })
-        warehouse_2 = self.env['stock.warehouse'].create({
+    def setUp(self):
+        super().setUp()
+
+        # Set two warehouses
+        self.warehouse_1 = self.env['stock.warehouse'].sudo().search([('company_id', '=', self.env.company.id)], limit=1)
```

Comment:
use data in test :thinking:

Reply:
I pushed an update on line 14 (`create_missing_warehouse`) to ensure such a warehouse exists

Reply:
What JKE probably meant, is more that the warehouse (being a data) might have been updated (added some rule/hierarchy or whatever) and could not be reliable, more than it could have been deleted.
Better to create a brand new one :+1:

Reply:
Yes but it will fail the test because, considering our fix, another warehouse will be returned by `_get_warehouse_available`
I first wanted to delete all existing warehouses but some SO already exist and are linked to some warehouses

Reply:
The test "just" needed to be ran on a new company then, so it is running in a ~standalone mode, not being impacted by the database state.